frontends/mit-open/src/pages/DashboardPage/SettingsPage.test.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react"
+import { SettingsPage } from "./SettingsPage"
+import { renderWithProviders, screen, within, user } from "@/test-utils"
+import { urls, setMockResponse, factories, makeRequest } from "api/test-utils"
+import type { LearningResourcesUserSubscriptionApiLearningResourcesUserSubscriptionCheckListRequest as CheckSubscriptionRequest } from "api"
+
+type SetupApisOptions = {
+  isAuthenticated?: boolean
+  isSubscribed?: boolean
+  subscriptionRequest?: CheckSubscriptionRequest
+}
+const setupApis = ({
+  isAuthenticated = false,
+  isSubscribed = false,
+  subscriptionRequest = {},
+}: SetupApisOptions = {}) => {
+  setMockResponse.get(urls.userMe.get(), {
+    is_authenticated: isAuthenticated,
+  })
+
+  const subscribeResponse = isSubscribed
+    ? factories.percolateQueries.percolateQueryList({ count: 5 }).results
+    : factories.percolateQueries.percolateQueryList({ count: 0 }).results
+  setMockResponse.get(
+    `${urls.userSubscription.check(subscriptionRequest)}`,
+    subscribeResponse,
+  )
+  const unsubscribeUrl = urls.userSubscription.delete(subscribeResponse[0]?.id)
+  setMockResponse.delete(unsubscribeUrl, subscribeResponse[0])
+  return {
+    unsubscribeUrl,
+  }
+}
+
+describe("SettingsPage", () => {
+  it("Renders user subscriptions in a list", async () => {
+    setupApis({
+      isAuthenticated: true,
+      isSubscribed: true,
+      subscriptionRequest: {},
+    })
+    renderWithProviders(<SettingsPage />)
+
+    const followList = await screen.findByTestId("follow-list")
+    expect(followList.children.length).toBe(5)
+  })
+
+  test("Clicking 'Unfollow' removes the subscription", async () => {
+    const { unsubscribeUrl } = setupApis({
+      isAuthenticated: true,
+      isSubscribed: true,
+      subscriptionRequest: {},
+    })
+    renderWithProviders(<SettingsPage />)
+
+    const followList = await screen.findByTestId("follow-list")
+    const unsubscribeButton = within(followList).getAllByText("Unfollow")[0]
```

Comment:
I try to use `getByRole*` for most things, since it effectively adds an assertion about the role in the accessibility treat. E.g., if we know it's an element with role link, it's almost certainly tab-navigable[^1]
```suggestion
    const unsubscribeButton = within(followList).getAllByRole("link", { name: "Unfollow" })[0]
```

[^1] I say "almost certainly" because it's probably an anchor tag. If someone did `<div role="link" />` (don't!), that would show up as a link in the browser's accessibility treat, but wouldn't be tab navigable.